tests/streamlit_ui_tests/utils.ts

```diff
@@ -48,19 +48,17 @@ export const getMitoFrameWithTypeCSV = async (page: Page): Promise<FrameLocator>
 }
   
 export const awaitResponse = async (page: Page): Promise<void> => {
-    // Wait at least 25 ms for the message to send
-    await page.waitForTimeout(200);
-    /*
-        Then, wait for:
-        1. Mito to finish processing the message -- the text "Processing" not being visible in the Mito loading indicator
-        2. Streamlit to finish executing the message -- the text "Running" not being visible in the Streamlit 
-        loading indicator in the top right of the streamlit app
-
-        Sometimes the loading indicator appears before the Running message, so only relying on "Running" means the 
-        test will continue running before the response is received, causing tests to fail.
-    */
-    await expect(page.getByText("Processing")).toHaveCount(0) 
-    await expect(page.getByText("Running")).toHaveCount(0) 
+
+    // Check if the message "Running" is visible, and wait if it is
+    if ((await page.locator('text=Running').count()) === 0) {
+        // Wait at least 25 ms for the message to send, as there is 
+        // a 25ms delay in the message sending in streamlit. We actually
+        // wait 500ms to be safe -- which is a bit of a cost, but 
```

Comment:
I think this is a good call. The biggest slow down by far is flakey tests, so I would rather wait a little bit extra each awaitResponse to avoid a bunch of reruns due to flakey tests

Reply:
Want to just add this justification to the end of this comment in the code